lms/services/assignment.py

```diff
@@ -101,7 +110,7 @@ def _get_copied_from_assignment(self, lti_params) -> Assignment | None:
 
         return None
 
-    def get_assignment_for_launch(self, request) -> Assignment | None:
+    def get_assignment_for_launch(self, request, course: Course) -> Assignment | None:
```

Comment:
We already now the course from the callers, pass it in at this time instead of creating a service dependency here.